source/broadband/webpa_replace.c

```diff
@@ -475,33 +492,38 @@ static int getWritableParams(char *paramName, char ***writableParams, int *param
     char l_Subsystem[MAX_DBUS_INTERFACE_LEN] = { 0 };
     componentStruct_t ** ppComponents = NULL;
     char *tempStr = NULL;
+    char *isRbus =NULL;
     char temp[MAX_PARAMETERNAME_LEN] = { 0 };
     //TODO: Coverity Requirement stack space exceeds 10000 bytes use 2048bytes forMAX_PARAMETERNAME_LEN.
     char *paramList[MAX_PARAMETERNAME_LEN] = { 0 };
 #if !defined(RDKB_EMU)
     strncpy(l_Subsystem, "eRT.",sizeof(l_Subsystem));
 #endif
     parameterInfoStruct_t **parameterInfo = NULL;
-    WalPrint("==================== Start of getWritableParams ==================\n");
+    
+    isRbus = getParameterValue(RBUS_ENABLE);
+    WalInfo("Rbus =%s\n", isRbus);
+    WalInfo("==================== Start of getWritableParams ==================\n");
     snprintf(dst_pathname_cr, sizeof(dst_pathname_cr),"%s%s", l_Subsystem, CCSP_DBUS_INTERFACE_CR);
     ret = CcspBaseIf_discComponentSupportingNamespace(bus_handle,dst_pathname_cr, paramName, l_Subsystem, &ppComponents, &size);
-    WalPrint("size : %d, ret : %d\n",size,ret);
+    WalInfo("size : %d, ret : %d\n",size,ret);
     if (ret == CCSP_SUCCESS && size == 1)
     {
         WalInfo("parameterName: %s, CompName : %s, dbusPath : %s\n", paramName, ppComponents[0]->componentName, ppComponents[0]->dbusPath);
         ret = CcspBaseIf_getParameterNames(bus_handle,ppComponents[0]->componentName, ppComponents[0]->dbusPath, paramName,1,&val_size,&parameterInfo);
-        WalPrint("val_size : %d, ret : %d\n",val_size,ret);
+        WalInfo("val_size : %d, ret : %d\n",val_size,ret);
         if(ret == CCSP_SUCCESS && val_size > 0)
         {
             cnt1 = 0;
             for(cnt = 0; cnt < val_size; cnt++)
             {
                 len = strlen(paramName);
-                if(parameterInfo[cnt]->writable == 1)
+                if(((strncmp(isRbus, "true", strlen("true")) == 0) && parameterInfo[cnt]->writable == 2) || 
```

Comment:
Add null check for isRbus . 
Can we avoid doing multiple string operations for each parameter in for loop, by adding null check and defining a flag variable at the beginning and use that flag to to decide if it is rbus/dbus mode .